### PR TITLE
implement the new V2 API changes

### PR DESCRIPTION
**Pull request by adorobis**
Depends on latest https://github.com/somm15/PyViCare repo

### CODE REVIEW

File: vicare/climate.py [hunk 1]
```diff
@@ -145,12 +145,14 @@ def update(self):
             else:
                 self._current_temperature = None
             self._current_program = self._api.getActiveProgram()
-
+            try:
```
Comment (crazyfx1) [inline vicare/climate.py:148]: We will need the try catch on every property, since the execution will stop on the first exception.

Comment (adorobis) [reply vicare/climate.py:148]: I guess so, I've just made this quick fix to make it work :)

File: vicare/climate.py [hunk 2]
```diff
@@ -145,12 +145,14 @@ def update(self):
...
             # The getCurrentDesiredTemperature call can yield 'error' (str) when the system is in standby
-            desired_temperature = self._api.getCurrentDesiredTemperature()
-            if desired_temperature == PYVICARE_ERROR:
+                desired_temperature = self._api.getCurrentDesiredTemperature()
+                if desired_temperature == PYVICARE_ERROR:
```
[the remark this answers is not in the document]
Comment (adorobis) [reply vicare/climate.py]: Removed. I'm just catching any exception now, should we rather catch the ones that pyvicare actually returns?

**Comment by oischinger** [comment]
Great job @adorobis and @crazyfx1 . I had the same change applied locally and it worked for me.
I guess we should wait until we have a release of PyVicare and then update the dependency in manifest.json before er merge.

**Comment by crazyfx1** [comment]
It's released now @oischinger  as v1.0.0.

Who is taking this change to HA?
 
I think following changes are still missing:
[] Try catch for all properties
[] Remove all references to PYVICARE_ERROR
[] As you said update manifest
[] Provide changelog for the HA PR ;)

**Comment by oischinger** [comment]
I'm integrating exception handling and will upload a PR here first before pushing upstream to Home Assistant Core.

**Comment by oischinger** [comment]
Since I cannot just push to this PR I created a new one: https://github.com/oischinger/ha_vicare/pull/38
I'll finish up the error handling today evening.

**Comment by adorobis** [comment]
I've added two new binary_sensors: DHW Pump Active and Charging Active.

**Comment by gburas** [comment]
@adorobis  Hi, can You also add heating.dhw.oneTimeCharge? (if i asked in the wrong place i'm sorry, i'm rookie on the github)

**Comment by adorobis** [comment]
> This is very chaotic :(

Sorry, couldn't resist to add sensors that are available in the new API (maybe were there already before but I didn't try to add them earlier). I'm done and waiting for the final merged version from oishinger :)

**Comment by DarkTempIar** [comment]
Is just me missing the sensors heating.solar.sensors.*? If you could add them too?!

**Comment by adorobis** [comment]
@oischinger what do we do with this PR? I have a few more sensors in the pipeline, working very well. Should I commit them here?

**Comment by crazyfx1** [comment]
You can't add them yet since current version on PyPi doesnt have the methods in it.

I like the enthusiasm in this project but I think the priority is now to get the current set of sensors out in HA and add additional stuff later.

**Comment by oischinger** [comment]
Yeah, we first need a pypi release again.
Actually I think we should contribute directly to Home Assistant core once the PyVicare 1.0.0 change is merged:
https://github.com/home-assistant/core/pull/53281

This repo here doesn't apply the same commit hooks, Github actions, etc

**Comment by oischinger** [comment]
@adorobis should I contribute those sensors to HA Core or do you want to create a PR?

**Comment by adorobis** [comment]
@oischinger feel free to contribute it there. I need to familiarize myself with contribution rules to the HA Core first.

**Comment by oischinger** [comment]
Hi @adorobis 
Since the big refactorings seem to be mostly over it would be good to rebase this one and contribute it when you find some time.

**Comment by adorobis** [comment]
I'm just wondering what is the best way to do it. Actually I have implemented those new sensors in the custom_component from scratch as it looks differently to what it was before this major refactoring. Should I contribute here or rather to the HA integration directly?

**Comment by oischinger** [comment]
If you already have made the changes to the custom component feel free to create the PR here. I can take care of pushing it upstream afterwards.

**Comment by adorobis** [comment]
Is your repo now in sync with the HA integration?

**Comment by oischinger** [comment]
Yes. I just synced it

**Comment by adorobis** [comment]
Not valid anymore due to major refactoring of the main component. Created new pull request with 4 new binary sensors: https://github.com/oischinger/ha_vicare/pull/73

**Pull request closed without merging**